lib/mayaUsd/ufe/UsdUndoCreateStageWithNewLayerCommand.cpp

```diff
@@ -0,0 +1,211 @@
+    // name "transform1" to "stage1". The number-suffix will be automatically incremented if
+    // necessary. The second rename operation renames it from "stageX" to "stage1". This doesn't do
+    // anything for the transform itself but it will adjust the number-suffix of the proxy shape
+    // according to the suffix of the transform, because they now share the common prefix "stage".
+    status = _createProxyShapeDagMod.renameNode(proxyShape, "stageShape1");
+    if (MFAIL(status)) {
+        return;
+    }
+    status = _createProxyShapeDagMod.renameNode(transformObj, "stage1");
+    if (MFAIL(status)) {
+        return;
+    }
+    status = _createProxyShapeDagMod.renameNode(transformObj, "stage1");
+    if (MFAIL(status)) {
+        return;
+    }
+
+    // Get the global `time1` object and its `outTime` attribute.
+    MSelectionList selection;
+    selection.add("time1");
+    MObject time1;
+    status = selection.getDependNode(0, time1);
+    if (MFAIL(status)) {
+        return;
+    }
+    MFnDependencyNode time1DepNodeFn(time1, &status);
+    if (MFAIL(status)) {
+        return;
+    }
+    MObject time1OutTimeAttr = time1DepNodeFn.attribute("outTime", &status);
+    if (MFAIL(status)) {
+        return;
+    }
+
+    // Get the `time` attribute of the newly created mayaUsdProxyShape.
+    MDagPath proxyShapeDagPath = MDagPath::getAPathTo(proxyShape, &status);
+    if (MFAIL(status)) {
+        return;
+    }
+    MFnDependencyNode proxyShapeDepNodeFn(proxyShapeDagPath.node(), &status);
+    if (MFAIL(status)) {
+        return;
+    }
+    MObject proxyShapeTimeAttr = proxyShapeDepNodeFn.attribute("time", &status);
+    if (MFAIL(status)) {
+        return;
+    }
+
+    // Connect `time1.outTime` to `proxyShapde.time`.
+    status
+        = _createProxyShapeDagMod.connect(time1, time1OutTimeAttr, proxyShape, proxyShapeTimeAttr);
+    if (MFAIL(status)) {
+        return;
+    }
+
+    // Execute the operations.
+    status = _createProxyShapeDagMod.doIt();
+    if (MFAIL(status)) {
+        return;
+    }
+    _createProxyShapeSuccess = true;
+
+    // Create a UFE scene item for the newly created mayaUsdProxyShape.
+    Ufe::Path proxyShapeUfePath = MayaUsd::ufe::dagPathToUfe(proxyShapeDagPath);
+    _insertedChild = Ufe::Hierarchy::createItem(proxyShapeUfePath);
+
+    // Refresh the cache of the stage map.
+    // When creating the proxy shape, the stage map gets dirtied and cleaned. Afterwards, the proxy
+    // shape is renamed. The stage map does not observe the Maya data model, so renaming does not
+    // dirty the stage map again. Thus, the cache is in an invalid state, where it contains the
+    // path of the proxy shape before it was renamed. Calling UsdStageMap::proxyShape() refreshes
+    // the cache. See comments within UsdStageMap::proxyShape() for more details.
+    g_StageMap.proxyShape(proxyShapeUfePath);
+}
+
+void UsdUndoCreateStageWithNewLayerCommand::undo()
```

Comment:
MAYA-128939 was moved further down in the priority and is not considered a blocker. It will be investigated after merging this PR if the error message becomes too frequent.